ext/datadog_profiling_native_extension/collectors_cpu_and_wall_time_worker.c

```diff
@@ -1067,19 +1075,21 @@ static VALUE _native_allocation_count(DDTRACE_UNUSED VALUE self) {
 // On big applications, path 1. is the hottest, since we don't sample every option. So it's quite important for it to
 // be as fast as possible.
 static void on_newobj_event(VALUE tracepoint_data, DDTRACE_UNUSED void *unused) {
-  // Update thread-local allocation count
-  if (RB_UNLIKELY(allocation_count == UINT64_MAX)) {
-    allocation_count = 0;
-  } else {
-    allocation_count++;
-  }
-
   struct cpu_and_wall_time_worker_state *state = active_sampler_instance_state; // Read from global variable, see "sampler global state safety" note above
 
   // This should not happen in a normal situation because the tracepoint is always enabled after the instance is set
   // and disabled before it is cleared, but just in case...
   if (state == NULL) return;
 
+  if (RB_UNLIKELY(state->allocation_counting_enabled)) {
```

Comment:
Is `RB_UNLIKELY` some kind of compiler optimisation macro to optimise resulting code for the false outcome?
And is `allocation_counting_enabled` unlikely because it is off by default and you don't expect many users to enable it?

Reply:
Yes and yes!

([Here's where the wrapper is defined](https://github.com/ruby/ruby/blob/82aee1a9467c0f1bd33eb0247c5a0a8b8b9a5049/include/ruby/backward/2/assume.h#L34-L50))

TBH I don't expect it to have a huge difference between having and not having this macro, but I liked using it in this situation to have more self-describing code in terms of what's the expected common path and the exception code.